File: android/biangbiang-ui/src/main/java/dev/veeso/biangbiangui/config/LanguageVariant.kt
package dev.veeso.biangbiangui.config

import dev.veeso.biangbiangui.protocols.Transliterator

/**
 * A single transliteration variant of a [LanguageProfile].
 * Mirrors iOS `LanguageVariant`.
 */
data class LanguageVariant(
    val id: String,
    val displayName: String,
    val transliterator: Transliterator,
    /** BCP-47 code for TTS; `null` hides the Listen control for this variant. */
    val ttsLanguageCode: String?,
    /** `false` hides the translation UI as data — no conditional code path. */
    val translatable: Boolean,
)

File: Sources/BiangBiangUI/Config/LanguageVariant.swift
public struct LanguageVariant: Sendable {
    public let id: String
    public let displayName: String
    public let transliterator: any Transliterator
    /// BCP-47 code for TTS; `nil` hides the Listen control for this variant.
    public let ttsLanguageCode: String?
    /// `false` hides the translation UI as data — no conditional code path.
    public let translatable: Bool

    public init(id: String, displayName: String, transliterator: any Transliterator,
                ttsLanguageCode: String?, translatable: Bool)
    {
        self.id = id; self.displayName = displayName
        self.transliterator = transliterator
        self.ttsLanguageCode = ttsLanguageCode
        self.translatable = translatable
    }
}
